### PR TITLE
Switch to Pyglet 2.1 or dev releases

### PR DESCRIPTION
**Pull request by pushfoo**
TL;DR: Get new pyglet bug fixes + immutable vecs for #2021

### Why?

Per Discord discussion with @Cleptomania, this change:

1. enables [backward-compatible `BasicSprite.scale` behavior](https://github.com/pythonarcade/arcade/pull/2021#issuecomment-1999386042) in #2021:
    ```py
   # Will work with scalars as it does now
   sprite_instance.scale *= 2.0
   # Cleaner & more efficient than setting individual scale components
   sprite_instance.scale *= 2.0, 3.0
   ```
2. seems to show it works fine according to preliminary tests
3. matches how we've never made a release that wasn't on a preview pyglet build

### Changes

1. Use `pyglet==2.1dev2`
2. Update references to `pyglet.canvas` to refer to `pyglet.display`
3. Use new layout syntax per 2.1 styles

## Testing Steps Taken

- [ ] Human spot-check of examples (GUI, Text, and Media) on
  - [x] Linux
  - [ ] Windows
  - [ ] Mac

- [ ] Run unit & integration tests on
  - [x] Linux
  - [ ] Windows
  - [ ] Mac

## Follow-up Work

* Update #2021 with `BasicSprite` changes
* Thorough example / functionality shakedown per [the 2.1 migration guide](https://pyglet.readthedocs.io/en/development/programming_guide/migration.html)
* (Optional) Make arcade match the x, y, z style of pyglet's text / etc?

### CODE REVIEW

**Comment by pushfoo** [comment]
TL;DR: Looks like CI failures are all unimportant but annoying type annotation issues.

In addition to pre-existing CI failures from the GUI, we also gain a few new ones due to 2.1 changes on pyglet's side:
* ~~The pyglet `.pyi` file doesn't appear to be updated to reflect `s/pyglet.canvas/pyglet.display/g` (PyCharm + pyright dislike this)~~ fixed as of 2.1dev2
* In pyglet, `Literal` used to annotate arguments where `str` is more appropriate :cry:

**Comment by pushfoo** [comment]
Ben just shipped pyglet 2.1dev2. I plan on updating this branch to that after I get back in < 2 hrs.

**Comment by pushfoo** [comment]
The recent commits:
1. Resolve rebase conflicts
2. Fix a lingering pyright issue from either the conflict resolution or pyglet 2.0 -> 2.1 changes
